src/utils.rs

```diff
@@ -1,6 +1,23 @@
 use std::future::Future;
 use std::pin::Pin;
 
+use crate::Error;
+
 /// An owned dynamically typed [`Future`] for use in cases where you can't
 /// statically type your result or need to add some indirection.
 pub(crate) type BoxFuture<'a, T> = Pin<Box<dyn Future<Output = T> + Send + 'a>>;
+
+/// Create a response from an [`Error`]. The response's status code will be
+/// retrieved from the error and the response body is set to the error message
+/// for non-500 errors.
+pub(crate) fn create_error_response(err: Error) -> http_types::Response {
+    let mut res = http_types::Response::new(err.status());
+    res.set_content_type(http_types::mime::PLAIN);
+    // Only send the message if it is a non-500 range error. All
+    // errors default to 500 by default, so sending the error
+    // body is opt-in at the call site.
+    if !res.status().is_server_error() {
+        res.set_body(err.to_string());
```

Comment:
I don't think all apps would want the same error response here. I'm not sure what the right solution is, but perhaps we need a way of adding an Option<tide::Error> to a Response or something like that.

Reply:
I agree, however, I've only extracted the current error handling into that utility method, to make it re-usable. So this PR is not changing the current error handling. As far as I remember from Discord, the current approach for custom error handling would be a custom middleware that intercepts errors and converts them into a response (basically similar to this one https://github.com/http-rs/tide/issues/525#issuecomment-634144680).